src/scss/src/foundation/_mixins.scss

```diff
@@ -1,16 +1,169 @@
+@use "sass:list";
 /*=============================================
 =                 #MIXINS                     =
 =============================================*/
 
-
-
 /**
  *
  * Helpful reusable functions-like code that will be used to implement
  * styles for components
  *
  */
 
+/*=============================================
+ =                @media rules                =
+ =============================================*/
+
+@mixin tablet {
+    @media (min-width: map-get($breakpoints, "tablet")) {
+        @content;
+    }
+}
+
+@mixin desktop {
+    @media (min-width: map-get($breakpoints, "desktop")) {
+        @content;
+    }
+}
+
+@mixin mobile {
+    @media (min-width: map-get($breakpoints, "mobile")) {
+        @content;
+    }
+}
+
+/*=============================================
+=                 Spacings                    =
+=============================================*/
+
+/**
+ * use it set up margin or padding 
+ * 
+ * @include spacing(margin, 'lg', 'xxs') - will set up margin vertical for 32px and horizontal for 8px 
+ */
+
+@mixin spacing($property, $values...) {
+    $valid-properties: margin, padding;
+
+    @if not list.index($valid-properties, $property) {
+        @error "#{$property} is not a valid property. Expected one of #{$valid-properties}.";
+    }
+
+    $propertyValues: ();
 
+    @each $value in $values {
+        $spacingValue: map-get($spacing, $value);
+        $propertyValues: append($propertyValues, $spacingValue);
+    }
 
+    #{$property}: $propertyValues;
+}
```

Comment:
This method has a bit hard to understand interface. While reading example `@include spacing(margin, 'lg', 'xxs')` I did not get the point what this mixin is actually doing. I suggest splitting this mixin into multiple mixins like following:

```scss
@mixin margin($value);
@mixin margin-push($direction: left, right, top, bottom, $value);
@mixin margin-x($value); // (use margin right and margin left here because we want to create as modular code as possible)
@mixin margin-y($value) // same here;
```

Same for padding

Reply:
I think it's pretty self-explanatory, it's the same as you'd write default margin with 1/2/3/4 parameter, but instead of `rem/px` this mixin accepts our custom breakpoints only

Reply:
No, i persuade that my approach will be more readable and easier to understand

### Example 1:

For example you want to add `margin` `xxs` on the `left` and `lg` on the `bottom`:

Current implemention:

```scss
@include spacing(margin, 'none', 'none', 'lg', "xxs");
```

Suggested implementation:

```scss
@include margin-push($left, 'xxs');
@include margin-push($bottom, 'lg');
```

### Example 2:

I want to add horizontal `padding` of `lg`:

Current implemention:

```scss
@include spacing(padding, 'none', 'lg');
```

Suggested implemention:

```scss
@include padding-x('lg');
```